spec/default_spec.rb

```diff
@@ -3,9 +3,12 @@
 describe 'threatstack::default' do
   context 'single-ruleset-test' do
     let(:chef_run) do
-      ChefSpec::SoloRunner.new do |node|
-        node.set['threatstack']['deploy_key'] = 'ABCD1234'
-      end.converge(described_recipe)
+      ChefSpec::SoloRunner.new.converge(described_recipe)
+    end
+
+    before do
+      contents = { 'deploy_key' => 'ABCD1234' }
+      expect(Chef::EncryptedDataBagItem).to receive(:load).with('threatstack', 'api_keys').and_return(contents)
```

Comment:
This should probably be `allow(Chef::...` in case someone wants to test a codepath that doesn't load the data bag.